src/main/kotlin/uk/gov/dwp/dataworks/services/TaskDeploymentService.kt

```diff
@@ -107,29 +113,83 @@ class TaskDeploymentService {
                 .build()
         val jupyterOverride= ContainerOverride.builder()
                 .name("jupyterHub")
-                .environment(userName, emrClusterHostName)
+                .environment(userNamePair, emrClusterHostName)
                 .cpu(jupyterCpu)
                 .memory(jupyterMemory)
                 .build()
         val guacDOverride = ContainerOverride.builder()
                 .name("guacd")
-                .environment(userName)
+                .environment(userNamePair)
                 .build()
 
         val overrides = TaskOverride.builder()
                 .containerOverrides(guacDOverride, chromeOverride, jupyterOverride)
+                .taskRoleArn(createTaskRoleOverride(listOfAdditionalPermissions, userName))
                 .build()
 
         return RunTaskRequest.builder()
                 .cluster(ecsClusterName)
                 .launchType("EC2")
                 .overrides(overrides)
-                .taskDefinition("orchestration-service-ui-service")
+                .taskDefinition("orchestration-service-analytical-workspace")
                 .build()
     }
-}
 
+    fun createTaskRoleOverride(list: List<String>, userName: String): String{
+        val iamClient = IamClient.builder().region(Region.AWS_GLOBAL).build()
+        fun additionalPermissions(): String{
+            val listToString = StringBuilder()
+            if(list.isNotEmpty()) for (i in list) listToString.append(",\"$i\"")
+            return listToString.toString()
+        }
+        val assumeRolePolicyDocument = "{" +
+        "  \"Version\": \"2012-10-17\"," +
+                "  \"Statement\": [" +
+                "    {" +
+                "        \"Effect\": \"Allow\"," +
+                "        \"Action\": [" +
+                "           \"sts:AssumeRole\"" +
+                "       ]," +
+                "       \"Principal\": {"  +
+                "           \"Service\": ["  +
+                "           \"ecs-tasks.amazonaws.com\"" +
+                "           ]" +
+                "        }" +
+                "     }" +
+                "   ]" +
+                "}"
+
+        val taskRolePolicyDocument = "{" +
+                "  \"Version\": \"2012-10-17\"," +
+                "  \"Statement\": [" +
+                    "     {" +
+                    "        \"Effect\": \"Allow\"," +
+                    "        \"Action\": [" +
+                    "            \"ecr:BatchCheckLayerAvailability\"," +
+                    "            \"ecr:GetDownloadUrlForLayer\"," +
+                    "            \"ecr:BatchGetImage\"," +
+                    "            \"logs:CreateLogStream\"," +
+                    "            \"logs:PutLogEvents\"" +
+                                additionalPermissions() +
+                    "       ]," +
+                    "       \"Resource\": \"*\"" +
+                    "      }" +
+                    "   ]" +
+                    "}"
```

Comment:
I think loading these via a resources file would be a much nicer solution, however can be done at a later date. For now we can leave this as is

Reply:
Sounds good, I will resolve this and we can pick it up in future as suggested.